addons/website_mail/controllers/main.py

```diff
@@ -47,111 +46,90 @@ def _message_post_helper(res_model='', res_id=None, message='', token='', token_
+            record.check_access_rule('read')
             # add partner to session
             request.session['partner_id'] = partner_ids[0]
-            _object.message_subscribe(cr, SUPERUSER_ID, [_id], partner_ids, context=context)
+            record.sudo().message_subscribe(partner_ids)
             return True
 
     @http.route(['/website_mail/is_follower'], type='json', auth="public", website=True)
     def call(self, model, id, **post):
-        id = int(id)
-        cr, uid, context = request.cr, request.uid, request.context
-
-        partner_obj = request.registry.get('res.partner')
-        users_obj = request.registry.get('res.users')
-        obj = request.registry.get(model)
-
-        partner_id = None
-        public_id = request.website.user_id.id
-        if uid != public_id:
-            partner_id = users_obj.browse(cr, SUPERUSER_ID, uid, context).partner_id
+        user = request.env.user
+        partner = None
+        public_user = request.website.user_id
+        if user != public_user:
+            partner = request.env.user.partner_id
         elif request.session.get('partner_id'):
-            partner_id = partner_obj.browse(cr, SUPERUSER_ID, request.session.get('partner_id'), context)
-        email = partner_id and partner_id.email or ""
+            partner = request.env['res.partner'].sudo().browse(request.session.get('partner_id'))
 
         values = {
-            'is_user': uid != public_id,
-            'email': email,
+            'is_user': user != public_user,
+            'email': partner.email if partner else "",
             'is_follower': False,
             'alias_name': False,
         }
 
-        if not obj:
-            return values
-        obj_ids = obj.exists(cr, SUPERUSER_ID, [id], context=context)
-        if obj_ids:
-            if partner_id:
-                values['is_follower'] = len(
-                    request.registry['mail.followers'].search(
-                        cr, SUPERUSER_ID, [
-                            ('res_model', '=', model),
-                            ('res_id', '=', obj_ids[0]),
-                            ('partner_id', '=', partner_id.id)
-                        ], context=context)) == 1
+        record = request.env[model].sudo().browse(int(id)).exists()
+        if record and partner:
+            values['is_follower'] = request.env['mail.followers'].search_count([
+                        ('res_model', '=', model),
+                        ('res_id', '=', record.id),
+                        ('partner_id', '=', partner.id)
+                    ]) == 1
         return values
 
     @http.route(['/website_mail/post/json'], type='json', auth='public', website=True)
     def chatter_json(self, res_model='', res_id=None, message='', **kw):
-        res_id = int(res_id)
         try:
-            msg = _message_post_helper(res_model, res_id, message, **kw)
+            msg = _message_post_helper(res_model, int(res_id), message, **kw)
```

Comment:
As a good practice, more the data = {} outside of the try statement. We only want to protect the message post helper.